### PR TITLE
Allow to set an Alias and add tags to a CMK

### PR DESCRIPTION
**Pull request by DongJoo-clumio**
Testing:
- Install module with newly added parameters: Create alias and Add tags from user input
  ```
  ...

  # module.clumio_byok.aws_kms_alias.key_alias will be created
  + resource "aws_kms_alias" "key_alias" {
      + arn            = (known after apply)
      + id             = (known after apply)
      + name           = "alias/dongjoo"
      + name_prefix    = (known after apply)
      + target_key_arn = (known after apply)
      + target_key_id  = (known after apply)
    }

  # module.clumio_byok.aws_kms_key.multi_region_cmk_key[0] will be created
  + resource "aws_kms_key" "multi_region_cmk_key" {
      + arn                                = (known after apply)
      + bypass_policy_lockout_safety_check = false
      + customer_master_key_spec           = "SYMMETRIC_DEFAULT"
      + deletion_window_in_days            = 30
      + description                        = "The CMK for Clumio to use to encrypt backups"
      + enable_key_rotation                = true
      + id                                 = (known after apply)
      + is_enabled                         = true
      + key_id                             = (known after apply)
      + key_usage                          = "ENCRYPT_DECRYPT"
      + multi_region                       = true
      + policy                             = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "kms:*"
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = "arn:aws:iam::082615160027:root"
                        }
                      + Resource  = "*"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + rotation_period_in_days            = (known after apply)
      + tags                               = {
          + "key" = "value"
        }
      + tags_all                           = {
          + "key" = "value"
        }
    }

  ...

  Apply complete! Resources: 8 added, 0 changed, 0 destroyed.
  ```
  image
  image

- Install module without added parameters: Create default alias. No tags.
  ```
  # module.clumio_byok.aws_kms_alias.key_alias will be created
  + resource "aws_kms_alias" "key_alias" {
      + arn            = (known after apply)
      + id             = (known after apply)
      + name           = "alias/clumio-byok"
      + name_prefix    = (known after apply)
      + target_key_arn = (known after apply)
      + target_key_id  = (known after apply)
    }

  # module.clumio_byok.aws_kms_key.multi_region_cmk_key[0] will be created
  + resource "aws_kms_key" "multi_region_cmk_key" {
      + arn                                = (known after apply)
      + bypass_policy_lockout_safety_check = false
      + customer_master_key_spec           = "SYMMETRIC_DEFAULT"
      + deletion_window_in_days            = 30
      + description                        = "The CMK for Clumio to use to encrypt backups"
      + enable_key_rotation                = true
      + id                                 = (known after apply)
      + is_enabled                         = true
      + key_id                             = (known after apply)
      + key_usage                          = "ENCRYPT_DECRYPT"
      + multi_region                       = true
      + policy                             = jsonencode(
            {
              + Statement = [
                  + {
                      + Action    = "kms:*"
                      + Effect    = "Allow"
                      + Principal = {
                          + AWS = "arn:aws:iam::082615160027:root"
                        }
                      + Resource  = "*"
                    },
                ]
              + Version   = "2012-10-17"
            }
        )
      + rotation_period_in_days            = (known after apply)
      + tags_all                           = (known after apply)
    }
  ```
  
  image

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -79,6 +79,12 @@ resource "aws_kms_key" "multi_region_cmk_key" {
   enable_key_rotation      = true
   deletion_window_in_days  = var.deletion_window_in_days
   policy                   = data.aws_iam_policy_document.byok_policy_document.json
+  tags                     = var.key_tags
+}
+
+resource "aws_kms_alias" "key_alias" {
+  name = "alias/${var.key_alias_name}"
```
Comment (prakkarp) [inline main.tf:86]: Is the `alias/` prefix required?

**Review by prakkarp — APPROVED** [review]
LGTM

File: main.tf (hunk 1 above)
Comment (DongJoo-clumio) [reply main.tf:86]: Yes. Refer to: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kms_alias

**Comment by DongJoo-clumio** [comment]
Upgrade module from deploying version:
```
Terraform will perform the following actions:

  # module.clumio_byok.data.aws_iam_policy_document.role_policy_document will be read during apply
  # (depends on a resource or a module with changes pending)
 <= data "aws_iam_policy_document" "role_policy_document" {
      + id            = (known after apply)
      + json          = (known after apply)
      + minified_json = (known after apply)

      + statement {
          + actions   = [
              + "kms:CreateGrant",
              + "kms:DescribeKey",
              + "kms:ListGrants",
              + "kms:ListRetirableGrants",
              + "kms:PutKeyPolicy",
              + "kms:ReplicateKey",
              + "kms:RetireGrant",
              + "kms:RevokeGrant",
              + "kms:TagResource",
            ]
          + effect    = "Allow"
          + resources = [
              + "arn:aws:kms:*:082615160027:key/mrk-1eea0b05efc042b69201c52cc44a7da3",
            ]
        }
      + statement {
          + actions   = [
              + "kms:CreateKey",
            ]
          + effect    = "Allow"
          + resources = [
              + "*",
            ]
        }
    }

  # module.clumio_byok.aws_iam_role.byok_mgmt_role will be updated in-place
  ~ resource "aws_iam_role" "byok_mgmt_role" {
        id                    = "ClumioKMSRole"
        name                  = "ClumioKMSRole"
        tags                  = {}
        # (12 unchanged attributes hidden)

      - inline_policy {
          - name   = "ReplicaAndGrantManagement" -> null
          - policy = jsonencode(
                {
                  - Statement = [
                      - {
                          - Action   = [
                              - "kms:TagResource",
                              - "kms:RevokeGrant",
                              - "kms:RetireGrant",
                              - "kms:ReplicateKey",
                              - "kms:PutKeyPolicy",
                              - "kms:ListRetirableGrants",
                              - "kms:ListGrants",
                              - "kms:DescribeKey",
                              - "kms:CreateGrant",
                            ]
                          - Effect   = "Allow"
                          - Resource = "arn:aws:kms:*:082615160027:key/mrk-1eea0b05efc042b69201c52cc44a7da3"
                        },
                      - {
                          - Action   = "kms:CreateKey"
                          - Effect   = "Allow"
                          - Resource = "*"
                        },
                    ]
                  - Version   = "2012-10-17"
                }
            ) -> null
        }
      + inline_policy {
          + name   = "ReplicaAndGrantManagement"
          + policy = (known after apply)
        }
    }

  # module.clumio_byok.aws_kms_alias.key_alias will be created
  + resource "aws_kms_alias" "key_alias" {
      + arn            = (known after apply)
      + id             = (known after apply)
      + name           = "alias/dongjoo"
      + name_prefix    = (known after apply)
      + target_key_arn = (known after apply)
      + target_key_id  = "mrk-1eea0b05efc042b69201c52cc44a7da3"
    }

  # module.clumio_byok.aws_kms_key.multi_region_cmk_key[0] will be updated in-place
  ~ resource "aws_kms_key" "multi_region_cmk_key" {
        id                                 = "mrk-1eea0b05efc042b69201c52cc44a7da3"
      ~ tags                               = {
          + "key" = "value"
        }
      ~ tags_all                           = {
          + "key" = "value"
        }
        # (14 unchanged attributes hidden)
    }

Plan: 1 to add, 2 to change, 0 to destroy.

Apply complete! Resources: 1 added, 1 changed, 0 destroyed.
```
Before upgrade:
image
After upgrade and apply alias name:
image

**Pull request merged**